src/tests/jf_test_14.f90

```diff
@@ -0,0 +1,121 @@
+!*****************************************************************************************
+!> author: Jacob Williams
+!  date: 09/02/2015
+!
+! Module for the 14th unit test.
+
+module jf_test_14_mod
+
+    use json_module
+    use, intrinsic :: iso_fortran_env , only: error_unit, output_unit
+
+    implicit none
+    
+    character(len=*),parameter :: dir = '../files/inputs/'  !! working directory
+    character(len=*),parameter :: filename1 = 'test1.json'  !! the file to read
+    integer :: icount = 0  !! a count of the number of "name" variables found
+
+contains
+
+    subroutine test_14(error_cnt)
+    
+    !! Tests the traversal of a JSON structure
+    !!
+    !! It traverses the structure, looks for all "name" variables, and changes the name.
+    
+    implicit none
+
+    integer,intent(out) :: error_cnt !! report number of errors to caller
+    
+    type(json_value),pointer  :: json
+        
+    write(error_unit,'(A)') ''
+    write(error_unit,'(A)') '================================='
+    write(error_unit,'(A)') '   TEST 14'
+    write(error_unit,'(A)') '================================='
+    write(error_unit,'(A)') ''
+
+    error_cnt = 0
+    icount = 0 !number of name changes (should be 2)
+        
+    call json_initialize() !initialize the module
+    
+    call json_parse(dir//filename1,json)  !read the file
+    if (json_failed()) then
+        call json_print_error_message(error_unit)
+        error_cnt = error_cnt + 1
+    end if
+    
+    call json_traverse(json,rename) !traverse all nodes in the structure
+    if (json_failed()) then
+        call json_print_error_message(error_unit)
+        error_cnt = error_cnt + 1
+    end if
+
+    if (icount/=2) then
+        write(error_unit,'(A)') 'Error: should be 2 "name" variables in this file: '//filename1
+        error_cnt = error_cnt + 1
+    end if 
+    
+    if (error_cnt==0) then
+        write(output_unit,'(A)') ''
```

Comment:
I’d prefer everything other than `json_print` to write to stderr so that one day I can finish implementing the output validation with a JSON linter.